### PR TITLE
Prepare release 2.1

### PR DESCRIPTION
**Pull request by jsafrane**
/kind documentation

Update README and CHANGELOG for release 2.1

```release-note
NONE
```

### CODE REVIEW

**Comment by jsafrane** [comment]
cc @msau42, PTAL

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ This information reflects the head of this branch.
 
 | Compatible with CSI Version                                                                | Container Image             | Min K8s Version |
 | ------------------------------------------------------------------------------------------ | ----------------------------| --------------- |
-| [CSI Spec v1.0.0](https://github.com/container-storage-interface/spec/releases/tag/v1.0.0) | quay.io/k8scsi/csi-attacher | 1.15            |
+| [CSI Spec v1.2.0](https://github.com/container-storage-interface/spec/releases/tag/v1.2.0) | quay.io/k8scsi/csi-attacher | 1.15            |
```
Comment (msau42) [inline README.md]: Should we add recommended k8s version column since migration is beta in 1.17?

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ The following table reflects the head of this branch.
 | Feature       | Status  | Default | Description                                                                                   |
 | ------------- | ------- | ------- | --------------------------------------------------------------------------------------------- |
 | CSINode*      | Beta    | On      | external-attacher uses the CSINode object to get the driver's node name instead of the Node annotation. |
-| CSIMigration* | Alpha   | On      | [Migrating in-tree volume plugins to CSI](https://kubernetes.io/docs/concepts/storage/volumes/#csi-migration). |
+| CSIMigration* | Beta    | On      | [Migrating in-tree volume plugins to CSI](https://kubernetes.io/docs/concepts/storage/volumes/#csi-migration). |
```
Comment (msau42) [inline README.md]: CSINode is GA now, but we still use beta object for backwards compatibility. How do you want to note it? Or can we remove it altogether? Our min version is 1.15 which means that it should not be falling back on node annotation anymore

Comment (jsafrane) [reply README.md]: Have we ever removed `storage.k8s.io/v1beta1`? IMO we've always added new objects there (+v1alpha1) and we do not deprecate / remove it. We can keep using it and move to v1 in the next release, bumping just min. k8s version.

We should not use the node annotation, I filed https://github.com/kubernetes-csi/external-attacher/issues/203.

File: README.md (hunk 1 above)
Comment (jsafrane) [reply README.md]: added recommended version

**Comment by jsafrane** [comment]
Added recommended version + Saad's metrics to CHANGELOG

**Pull request merged**